models/model_choice.py

```diff
@@ -70,15 +68,17 @@ def net(net_params, inference=False):
         # whether one is subset/superset of the other.
         model.load_state_dict(chopped_dict, strict=False)
     elif model_name == 'deeplabv3_resnet101':
-        assert net_params['global']['number_of_bands'] == 3, msg
-        # pretrained on coco (21 classes)
-        coco_model = models.segmentation.deeplabv3_resnet101(pretrained=True, progress=True,
-                                                        num_classes=21, aux_loss=None)
-        model = models.segmentation.deeplabv3_resnet101(pretrained=False, progress=True,
-                                                        num_classes=num_classes, aux_loss=None)
-        chopped_dict = chop_layer(coco_model.state_dict(), layer_names=['classifier.4'])
-        del coco_model
-        model.load_state_dict(chopped_dict, strict=False)
+        try:
+            model = models.segmentation.deeplabv3_resnet101(pretrained=False, progress=True, in_channels=num_bands,
+                                                            num_classes=num_channels, aux_loss=None)
+        except:
+            assert num_bands==3, 'Edit torchvision scripts segmentation.py and resnet.py to build deeplabv3_resnet ' \
+                                 'with more or less than 3 bands'
```

Comment:
Load deeplabv3 with num_channels != 3. Requires modification in torchvision code.